catalyst_voices_packages/catalyst_cardano_serialization/lib/src/types.dart

```diff
@@ -0,0 +1,26 @@
+import 'package:cbor/cbor.dart';
+
+/// Specifies an amount of ADA in terms of lovelace.
+typedef Coin = int;
+
+/// Specifies on which network the code will run.
+enum NetworkId {
+  /// The production network
+  mainnet(magicId: 1),
+
+  /// The test network.
+  testnet(magicId: 0);
+
+  /// The magic protocol number acting as the identifier of the network.
+  final int magicId;
```

Comment:
Why do we have  `magicId`  and not to something like  `id` or `networkId`?

Reply:
I'll rename to id. Some CIP's call it a "network magic number", there are lots of resources that call it that way too, for instance: https://forum.cardano.org/t/cardano-cli-testnet-significance-of-integer-value-after-testnet-magic/71129.

In CIP-19 it's called `network tag` on the other hand:

![Screenshot 2024-05-09 at 15 53 03](https://github.com/input-output-hk/catalyst-voices/assets/166132265/7b304e02-e67d-434e-acd8-42543bffa234)

Reply:
We don't need to follow a bad naming convention.
Let's mention in the comment to the property that in some docs, this can be named like `network magic number.` etc. What do you think?

Reply:
I already did:
```
/// The magic protocol number acting as the identifier of the network.
```